redis_session.c

```diff
@@ -301,8 +281,11 @@ static int lock_acquire(RedisSock *redis_sock, redis_session_lock_status *lock_s
 
     /* Attempt to get our lock */
     for (i = 0; retries == -1 || i <= retries; i++) {
-        if (set_session_lock_key(redis_sock, cmd, cmd_len TSRMLS_CC) == SUCCESS) {
-            lock_status->is_locked = 1;
+        char *reply;
+        int reply_len;
+        if ((reply = redis_simple_cmd(redis_sock, cmd, cmd_len, &reply_len TSRMLS_CC)) != NULL) {
```

Comment:
Good simplification, but does currently not catch network failures.

The intention of this PR war to break the loop on network failures, which are characterized by:
* reply == NULL
* AND reply_len < 0

If this conditions occur, we should break the loop while setting lock_status->is_locked = 0

Reply:
The current implementation of redis_simple_cmd, does not differentiate between empty responses while reading the socket and socket write failures.

Therefore the function was adjusted in the initial PR:
https://github.com/phpredis/phpredis/pull/1421/files#diff-a8c2cac018b55b5883b8faad30efbfb4R154

Reply:
> If this conditions occur, we should break the loop while setting lock_status->is_locked = 0
Next line do this :)

Reply:
When network failure happens redis_simple_cmd will return NULL else we check the value and set `is_locked` appropriately

Reply:
> Next line does this :)

Yes, but the next line is never reached, because reply **==** NULL.
As the next line is only reached if reply **!=** NULL:
````
(reply = redis_simple_cmd(redis_sock, cmd, cmd_len, &reply_len TSRMLS_CC)) != NULL
````



> When network failure happens redis_simple_cmd will return NULL

That is true, but not the only case for NULL return.
I debugged the code and NULL gets returned in two cases:
1. Network failure
2. Locking key of another process is already present.

In case 1, we want to break the look immediately
In case 2, we want to sleep and retry

So without changing the return behavior of  redis_simple_cmd(), there is no way to differentiate between this two cases.

Reply:
In my mind `redis_simple_cmd` returts NULL only in case of network problem.

Reply:
@SkydiveMarius I looked at your PR once again and now see the difference.